internal/common/database/migrations_test.go

```diff
@@ -0,0 +1,46 @@
+package database
+
+import (
+	"testing"
+	"strconv"
+	"strings"
+	"io/fs"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadMigrations(t *testing.T) {
+	mockFiles := []string{
+		"migration1.sql",
+	}
+	mockFS := createMockFileSystem(mockFiles)
+	migrations, err := ReadMigrations(mockFS, "path")
+
+	// Checks the suffix of files and handles file naming format
+	for _, migration := range migrations {
+		assert.True(t, strings.HasSuffix(migration.name, "migration.sql"), "Invalid file format for %v", migration.name)
+	}
+	
+	// Checks slice format
+	expectedMigrations := []Migration{
+		{id: 0, name: "", sql: ""},
+	}
+	assert.Len(t, migrations, 1, "Unexpected slice length")
+
+	// Slice first value should be int
+	assert.IsType(t, expectedMigrations[0].id, migrations[0].id, "Should be of type int")
+	// Slice second value should be string
+	assert.IsType(t, expectedMigrations[0].name, migrations[0].name, "Should be of type string")
+	// Slice third value should be string
+	assert.IsType(t, expectedMigrations[0].sql, migrations[0].sql, "Should be of type string")
+
+	// Checks ID length
+	for _, migration := range migrations {
+		assert.True(t, len(strconv.Itoa(migration.id)) <= 5, "Invalid ID length for %v", migration.name)
+	}
+	assert.NoError(t, err, "Unexpected error in ReadMigrations")
+}
+
+/*
+func createMockFileSystem(files []string) fs.FS {
```

Comment:
Here you just need to declare a new struct type, and then have it implement the same method(s) that are defined for `fs.FS` - see https://pkg.go.dev/io/fs@go1.20.12

Reply:
Can you please elaborate this one?

Reply:
This would be just an instance of creating a classic Go interface implementation. You'd create a struct (for example, `type struct testFS { ... }`) and then create method(s) that implement the `fs.FS` interface - in this case, there's just one method, so the new function definition would be 
`func (myFS *testFS) Open(name string) (File, error) { /* return a fake File object */ }`.

If you search for `golang interfaces`, there are several good articles on the basics of implementing interfaces in Go, like https://golangbot.com/interfaces-part-1/ .

Reply:
got it...

Reply:
I have created struct `testFS` for `Mockfiles` handling